src/views/ResLife/components/RAView/index.jsx

```diff
@@ -0,0 +1,394 @@
+import { checkIfCheckedIn, submitCheckIn } from 'services/residentLife/RA_Checkin';
+import { preferredContact, PrefContactMethod } from 'services/residentLife/ResidentStaff';
+import { useUser } from 'hooks';
+import HousingBanner from '../ResidentView/components/HousingWelcome/Banner';
+
+const RAView = () => {
+  const [isCheckedIn, setCheckedIn] = useState(false);
+  const [open, setOpen] = useState(false);
+  const [confirmOpen, setConfirmOpen] = useState(false);
+  const [isChecked, setIsChecked] = useState(false);
+  const { profile } = useUser();
+  const [selectedContact, setSelectedContact] = useState('');
+  const [hallName, setHallName] = useState('');
+
+  const isMobile = useMediaQuery('(max-width:600px)');
+
+  // Fetch check-in status and initialize hall data
+  useEffect(() => {
+    const fetchData = async () => {
+      if (profile?.ID) {
+        try {
+          const isChecked = await checkIfCheckedIn(profile.ID);
+          setCheckedIn(isChecked);
+
+          if (profile.hall) {
+            setHallState((prevState) => ({
+              ...prevState,
+              [profile.hall]: true,
+              village: ['CON', 'GRA', 'RID', 'MCI'].includes(profile.hall),
+            }));
+          }
+        } catch (error) {
+          console.error('Error fetching data:', error);
+        }
+      }
+    };
+
+    fetchData();
+  }, [profile?.ID]);
+
+  //Auto set radio button in pref contact card to current pref method
+  useEffect(() => {
+    const fetchPreferredContact = async () => {
+      if (profile?.ID) {
+        try {
+          const contactPreference = await PrefContactMethod(profile.ID);
+          setSelectedContact(contactPreference?.PreferredContactMethod || '');
+        } catch (error) {
+          console.error('Error fetching preferred contact method:', error);
+        }
+      }
+    };
+
+    fetchPreferredContact();
+  }, [profile?.ID]);
+
+  const hallCodeToNameMap = {
+    BRO: 'Bromley',
+    CHA: 'Chase',
+    EVN: 'Evans',
+    FER: 'Ferrin',
+    FUL: 'Fulton',
+    NYL: 'Nyland',
+    TAV: 'Tavilla',
+    WIL: 'Wilson',
+    CON: 'Conrad',
+    GRA: 'Grace',
+    RID: 'Rider',
+    MCI: 'MacInnis',
+    village: 'The Village',
+  };
+
+  const handleConfirm = () => {
+    setConfirmOpen(true);
+
+    let tempName = '';
+
+    for (let hall in hallState) {
+      if (hallState[hall] && hall !== 'village') {
+        // Map hall codes to names using the mapping object
+        const hallName = hallCodeToNameMap[hall];
+        tempName = tempName ? `${tempName}, ${hallName}` : hallName;
+      }
+    }
+
+    setHallName(tempName);
+  };
+
+  const handleSubmit = async () => {
+    const selectedHallCodes = Object.keys(hallState).filter(
+      (hall) => hallState[hall] && hall !== 'village',
+    ); //exclude village for checkin
+
+    if (!profile?.ID || selectedHallCodes.length === 0) {
+      alert('Please select a hall and ensure profile information is loaded before checking in.');
```

Comment:
The `alert` function is very old browser UI that most users don't appreciate. Snackbars are the preferred way of sending timely feedback to users these days, as seen in e.g. the IDUploader page.